modules/gdscript/gdscript_compiler.cpp

```diff
@@ -131,16 +135,39 @@ GDScriptDataType GDScriptCompiler::_gdtype_from_datatype(const GDScriptParser::D
 						script = script->subclasses[names.back()->get()];
 						names.pop_back();
 					}
-					result.kind = GDScriptDataType::GDSCRIPT;
 					result.script_type = script.ptr();
 					result.native_type = script->get_instance_base_type();
-					result.builtin_type = p_datatype.builtin_type;
 				} else {
```

Comment:
To make the code easier to understand, it could be good to add a short comment that explains what the `else` case is here (i.e. what are we working on if not "Local class." as above).